cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,217 @@
+import { TEST_REPO_NAME } from "../fixtures/constants"
+
+const visitHomepage = () => {
+  cy.visit(`/sites/${TEST_REPO_NAME}/homepage`)
+}
+
+const NOTIFICATION_BAR_PREVIEW_SELECTOR = "div[id='notification-bar']"
+const NOTIFICATION_INPUT_SELECTOR = 'input[id="site-notification"]'
+
+const HERO_BUTTON_INPUT_SELECTOR = 'input[id*="hero-button"]'
+const HERO_BUTTON_PREVIEW_SELECTOR = 'div[class*="search-button"]'
+
+const HERO_SECTION_HIGHLIGHTS_SELECTOR = 'input[id="radio-highlights"]'
+const HERO_SECTION_DROPDOWN_SELECTOR = 'input[id="radio-dropdown"]'
+
+const ADD_SECTION_DROPDOWN_SELECTOR = 'select[id="section-new"]'
+
+const getAddSectionButtonSelector = (): "@addSectionButton" => {
+  cy.contains("div", "Add a new section").find("button").as("addSectionButton")
+
+  return "@addSectionButton"
+}
+
+const getPreviewSectionSelectorByTitle = (title: string): `@${string}` => {
+  cy.contains("section", title).as(`section-${title}`).should("exist")
+  return `@section-${title}`
+}
+
+const SECTION_DROPDOWN_OPTIONS = {
+  DEFAULT: "--Choose a new section--",
+  RESOURCES: "Resources",
+  INFOBAR: "Infobar",
+  INFOPIC: "Infopic",
+}
+
+const toggleSectionDropdown = (section: string) => {
+  cy.contains(section).next().click()
+}
+
+const timings = {
+  outOfTheWay: 0.2,
+  // greater than the out of the way time
+  // so that when the drop ends everything will
+  // have to be out of the way
+  minDropTime: 0.33,
+  maxDropTime: 0.55,
+}
+
+const keyCodes = {
+  space: 32,
+  arrowDown: 40,
+}
+
+const REACT_DRAG_HANDLE_PROPERTY = "data-rfd-drag-handle-draggable-id" as const
+
+const getHandleSelector = (draggableId?: string) => {
+  if (draggableId) {
+    return `[${REACT_DRAG_HANDLE_PROPERTY}="${draggableId}"]`
+  }
+  return `[${REACT_DRAG_HANDLE_PROPERTY}]`
+}
```

Comment:
Could we consider moving this selector (and the associated reordering) into a utils file? Nav bar/Contact us also make use of this and are similarly missing tests

Reply:
can! i just left it here first because the other tests aren't up